Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp

```diff
@@ -166,30 +166,20 @@ void MediaSourcePrivateRemote::setTimeFudgeFactor(const MediaTime& fudgeFactor)
     gpuProcessConnection->connection().send(Messages::RemoteMediaSourceProxy::SetTimeFudgeFactor(fudgeFactor), m_identifier);
 }
 
-Ref<MediaTimePromise> MediaSourcePrivateRemote::waitForTarget(const WebCore::SeekTarget& target)
-{
-    ASSERT_NOT_REACHED();
-    return MediaTimePromise::createAndReject(PlatformMediaError::LogicError);
-}
-
 void MediaSourcePrivateRemote::proxyWaitForTarget(const WebCore::SeekTarget& target, CompletionHandler<void(MediaTimePromise::Result&&)>&& completionHandler)
 {
-    if (!m_client)
-        return completionHandler(makeUnexpected(PlatformMediaError::ClientDisconnected));
-    m_client->waitForTarget(target)->whenSettled(RunLoop::current(), WTFMove(completionHandler));
-}
-
-Ref<MediaPromise> MediaSourcePrivateRemote::seekToTime(const MediaTime& time)
-{
-    ASSERT_NOT_REACHED();
-    return MediaPromise::createAndReject(PlatformMediaError::LogicError);
+    if (RefPtr client = this->client())
+        client->waitForTarget(target)->whenSettled(RunLoop::current(), WTFMove(completionHandler));
+    else
+        completionHandler(makeUnexpected(PlatformMediaError::ClientDisconnected));
```

Comment:
I tend to prefer if(!client) return completionHandler style

Reply:
but that requires one more line and making `client` scope longer than it needs to be.